### PR TITLE
image decompress timer

### PR DESCRIPTION
**Pull request by chriselion**
We want to gather timings on how much we currently spend decompressing during visual learning.

### CODE REVIEW

File: ml-agents-envs/mlagents/envs/brain.py [hunk 1]
```diff
@@ -161,8 +162,9 @@ def process_pixels(image_bytes: bytes, gray_scale: bool) -> np.ndarray:
         :param image_bytes: input byte array corresponding to image
         :return: processed numpy array of observation from environment
         """
-        image_bytearray = bytearray(image_bytes)
-        image = Image.open(io.BytesIO(image_bytearray))
+        with hierarchical_timer("image_decompress"):
+            image_bytearray = bytearray(image_bytes)
+            image = Image.open(io.BytesIO(image_bytearray))
```
Comment (chriselion) [inline ml-agents-envs/mlagents/envs/brain.py]: Actually, I might need to add the next line under the timer too. According to the [docs](https://pillow.readthedocs.io/en/4.2.x/reference/Image.html#PIL.Image.open), `Image.open()` 
> is a lazy operation; this function identifies the file, but the file remains open and the actual image data is not read from the file until you try to process the data (or call the load() method).

Comment (surfnerd) [reply ml-agents-envs/mlagents/envs/brain.py]: 👍

**Comment by chriselion** [comment]
Just a note on the lazy loading. Here's timings from the same run of GridWorld with and without explicitly calling Image.load():

`without load()`
``` json
{
  "name": "BrainInfo.process_pixels",
  "total": 3.584165224000029,
  "count": 10002,
  "is_parallel": true,
  "self": 2.7557900760001024,
  "children": [
    {
      "name": "image_decompress",
      "total": 0.8283751479999264,
      "count": 10002,
      "is_parallel": true,
      "self": 0.8283751479999264
    }
  ]
}
```

`with load()`
``` json
{
  "name": "BrainInfo.process_pixels",
  "total": 3.646127201000195,
  "count": 10001,
  "is_parallel": true,
  "self": 0.9565701650002723,
  "children": [
    {
      "name": "image_decompress",
      "total": 2.689557035999923,
      "count": 10001,
      "is_parallel": true,
      "self": 2.689557035999923
    }
  ]
}
```
So total time is about the same (within error bars) but forcing the load shifts the time into the `image_decompress` block (where it belongs).

**Comment by chriselion** [comment]
BTW, total train time (wallclock) was about 36.78seconds for those runs.

**Pull request merged**